java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -86,7 +86,82 @@ private predicate sourceModelCsv(string row) {
       // SocketGetInputStreamMethod
       "java.net;Socket;false;getInputStream;();;ReturnValue;remote",
       // BeanValidationSource
-      "javax.validation;ConstraintValidator;true;isValid;;;Parameter[0];remote"
+      "javax.validation;ConstraintValidator;true;isValid;;;Parameter[0];remote",
+      // SpringMultipartRequestSource
+      "org.springframework.web.multipart;MultipartRequest;true;getFile;(String);;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartRequest;true;getFileMap;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartRequest;true;getFileNames;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartRequest;true;getFiles;(String);;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartRequest;true;getMultiFileMap;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartRequest;true;getMultipartContentType;(String);;ReturnValue;remote",
+      // SpringMultipartFileSource
+      "org.springframework.web.multipart;MultipartFile;true;getBytes;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getContentType;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getInputStream;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getName;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getOriginalFilename;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getResource;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getSize;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;isEmpty;();;ReturnValue;remote",
```

Comment:
This line looks added by mistake. I don't think `isEmpty` should be included.

Reply:
Previously only `get%` methods were covered, but now that I had to add the methods one by one, I also added `isEmpty`. Don't you think it's analogous to `getSize`?

Reply:
Somewhat, yes. Let's remove `getSize` as well. I'd say that a boolean can't be tainted (it's simply too few bits). A long can be considered tainted in some contexts, but this is mostly relevant for C/C++ and not so often for Java. And regardless I don't think measuring the size of some tainted data should be considered tainted.